src/java/org/apache/cassandra/config/YamlConfigurationLoader.java

```diff
@@ -163,21 +184,64 @@ private static void maybeAddSystemProperties(Object obj)
         if (CassandraRelevantProperties.CONFIG_ALLOW_SYSTEM_PROPERTIES.getBoolean())
         {
             java.util.Properties props = System.getProperties();
-            Map<String, String> map = new HashMap<>();
-            for (String name : props.stringPropertyNames())
+            Map<String, Object> map = new HashMap<>();
+            for (String originalKey : props.stringPropertyNames())
             {
-                if (name.startsWith(SYSTEM_PROPERTY_PREFIX))
+                if (originalKey.startsWith(SYSTEM_PROPERTY_PREFIX))
                 {
-                    String value = props.getProperty(name);
-                    if (value != null)
-                        map.put(name.replace(SYSTEM_PROPERTY_PREFIX, ""), value);
+                    String value = props.getProperty(originalKey);
+                    String configKey = originalKey.replace(SYSTEM_PROPERTY_PREFIX, "");
+                    if (OVERRIDABLE_CONFIG_NAMES.contains(configKey) && value != null && !map.containsKey(configKey))
```

Comment:
if a config is defined that isn't in this map, should we at least log to tell you?  the prefix is very specific so the intent is clear that it was trying to override